framework/sessions/__init__.py

```diff
@@ -165,14 +164,8 @@ def before_request():
         if not throttle_period_expired(user_session.created, settings.OSF_SESSION_TIMEOUT):
             # Update date last login when making non-api requests
             if user_session.data.get('auth_user_id') and 'api' not in request.url:
-                OSFUser = apps.get_model('osf.OSFUser')
-                (
-                    OSFUser.objects
-                    .filter(guids___id__isnull=False, guids___id=user_session.data['auth_user_id'])
-                    # Throttle updates
-                    .filter(Q(date_last_login__isnull=True) | Q(date_last_login__lt=timezone.now() - dt.timedelta(seconds=settings.DATE_LAST_LOGIN_THROTTLE)))
-                ).update(date_last_login=timezone.now())
-            set_session(user_session)
+                update_user_date_last_login(user_session)
+                set_session(user_session)
```

Comment:
Did you mean to move `set_session(user_session)` into the `if user_session.data...` if block?

Reply:
Good catch!